doc/api/errors.md

```diff
@@ -1204,6 +1204,17 @@ An invalid `options.protocol` was passed.
 Both `breakEvalOnSigint` and `eval` options were set in the REPL config, which
 is not supported.
 
+<a id="ERR_INVALID_RETURN_PROPERTY"></a>
+### ERR_INVALID_RETURN_PROPERTY
+
+Thrown in case a function option does not return an expected property type.
+
+<a id="ERR_INVALID_RETURN_PROPERTY_STRING"></a>
+### ERR_INVALID_RETURN_PROPERTY_STRING
+
+Thrown in case a function option does not return an expected string property
+type.
```

Comment:
I am not sure why strings got special handled while other types not. I recommend to change this to: `ERR_INVALID_RETURN_PROPERTY_VALUE`.
The current description is definitely wrong as the usage is not about the returned type but about the returned value (at least when looking at the actual implementation).

Reply:
Agreed on the semantics, but the thing is ideally the current `ERR_INVALID_RETURN_VALUE` error should then be `ERR_INVALID_RETURN_TYPE` as that is what it is really checking. So I was trying to remain consistent with that. I think `ERR_INVALID_RETURN_PROPERTY_VALUE` isn't suitable for this reason. `type` in the description is referring to "types of strings" as in categories / enums.

Reply:
I agree that the `ERR_INVALID_RETURN_VALUE` could be improved. However the return checks are not common and could be consolidated to always be a `ERR_INVALID_RETURN_VALUE`, just with a more specific error message. IMHO differentiating the type error from the value error is not useful for users.

That all aside: I am definitely against having a error code that is specific for `property strings`. So at least that has to be more generalized out of my perspective.

Reply:
> IMHO differentiating the type error from the value error is not useful for users.

See my example response in the last comment for why I definitely feel this is a useful distinction in the error message.

In terms of naming, I've switched them around to use `ERR_INVALID_RETURN_PROPERTY_VALUE` to be consistent with return value, then made the new one just `ERR_INVALID_RETURN_PROPERTY`.